src/core/qgsziputils.cpp

```diff
@@ -84,17 +85,20 @@ bool QgsZipUtils::unzip( const QString &zipFilename, const QString &dir, QString
         char *buf = new char[len];
         if ( zip_fread( file, buf, len ) != -1 )
         {
-          QFileInfo newFile( QDir( dir ), QString( stat.name ) );
           QString fileName( stat.name );
-          bool isDirectory = fileName.lastIndexOf( "/" ) == fileName.length() - 1;
+          QFileInfo newFile( QDir( dir ), fileName );
+          bool isDirectory = fileName.lastIndexOf( '/' ) == fileName.length() - 1;
```

Comment:
I still think that it's much simpler and clearer with:
bool isDirectory = fileName.endsWith( '/' );